python/lsst/pipe/tasks/isolatedStarAssociation.py

```diff
@@ -426,7 +426,8 @@ def _match_primary_stars(self, primary_bands, star_source_cat):
                                                self.config.match_radius/3600.)
                 # Any object with a match should be removed.
                 match_indices = np.array([i for i in range(len(idx)) if len(idx[i]) > 0])
-                band_cat = np.delete(band_cat, match_indices)
+                if len(match_indices) > 0:
+                    band_cat = np.delete(band_cat, match_indices)
```

Comment:
Worth adding a log message?

Reply:
This isn't a problem, it's fine for there to be no overlaps especially at the edges of things.